server.py

```diff
@@ -372,6 +372,13 @@ async def create_cluster(  # pylint: disable=too-many-arguments,too-many-positio
             description="The CPU architecture for the cluster. Defaults to 'x86_64' if not specified. Valid options are: x86_64, aarch64, arm64, ppc64le, s390x.",
         ),
     ] = "x86_64",
+    platform: Annotated[
+        str | None,
```

Comment:
Not a huge deal but this is a good candidate for a `Literal` type since you only have 4 valid non-None values.

See https://typing.python.org/en/latest/spec/literal.html